docs/docs/resources/architecture/components-hierarchy.md

```diff
@@ -0,0 +1,12 @@
+```mermaid
+classDiagram
+    BaseDefaultsComponent <|-- PipelineComponent
+    PipelineComponent <|-- KafkaConnector
+    PipelineComponent <|-- KubernetesApp
+    KafkaConnector <|-- KafkaSourceConnector
+    KafkaConnector <|-- KafkaSinkConnector
+    KubernetesApp <|-- KafkaApp
+    KafkaApp <|-- StreamsApp
+    KafkaApp <|-- ProducerApp
+```
+<p style="text-align: center;"><i>KPOps component hierarchy</i></p>
```

Comment:
I know that we are not doing standard UML here. But I don't understand why two empty rows are beneath each component. I suggest you keep it a bit more standard and use the component notion in UML. Or just use a square with no empty rows:
https://www.smartdraw.com/component-diagram/#:~:text=A%20component%20diagram%2C%20also%20known,is%20covered%20by%20planned%20development.

Reply:
It's because it is a UML class diagram and I couldn't make mermaid remove the empty rows for params and methods. [Unfortunately Mermaid doesn't yet support component graphs.](https://github.com/mermaid-js/mermaid/issues/1462).

Do you think I should put a [drawing ](https://github.com/bakdata/kpops/commit/d4a80cd616e97db43407dac15b5f1465c8074403#diff-e0459dd789dd26eb66a2514c8b9d8b894a850a3c16ad68989d4404b69231a27e)there instead? I drew it already because I disliked the Mermaid one, but maintenance and expanding the docs is likely gonna suck a lot.

Reply:
Okay, good to know. No, I would have it like this for now if it is too much effort to change it.

Reply:
Haha I already put in the effort, you can see it in [this commit](https://github.com/bakdata/kpops/commit/d4a80cd616e97db43407dac15b5f1465c8074403#diff-62c7ebdd73b1f25bcf499e1e8fd67713027e07f45f15cb9a31742a8cc7c8b088). It's more about what the future holds. I could just revert to it and make use of mermaid later when it better fits our needs?

Reply:
> > > Could you please configure the links so they can be opened in a new tab if the user clicks them?
> > 
> > 
> > imo this should be part of the user's browser settings. Personally, I am really not a fan of forcing everything to open in a new tab.
> 
> I share your preference @disrupted. I actually searched guidelines for that now, I was curious if there is a standard for that.
> 
> [The most formal thing I found.](https://www.w3.org/TR/WCAG20-TECHS/G200.html)

Okay, we can keep all the links that related to the document to be opened in the same window, but prefer links that lead to another website (i.e., GitHub) to be opened in a new window/tab.